convert2rhel/unit_tests/checks_test.py

```diff
@@ -473,31 +457,153 @@ def test_check_tainted_kmods(monkeypatch, command_return, expected_exception):
     )
     if expected_exception:
         with pytest.raises(expected_exception):
-            check_tainted_kmods()
+            checks.check_tainted_kmods()
     else:
-        check_tainted_kmods()
+        checks.check_tainted_kmods()
+
+
+class EFIBootInfoMocked():
```

Comment:
The interface of this object is completely detached from the original one. Is it possible to inherit from the origin one and just override some functionality?

Reply:
maybe Mock(autospec=True)?

Reply:
What is the original one? Can you elaborate on it?

Reply:
The original one is the one we mock - class  EfiBootInfo (https://github.com/oamg/convert2rhel/pull/223/files#diff-3a7eda59c836f39329b256971a1603cf9345a87f7abc30d6c681b5da961967b1R258). So using for example autospeccing (https://docs.python.org/3/library/unittest.mock.html#autospeccing) might improve the quality of the test and make it more useful. I think what we can do here is something which shown at the end of the chapter:
```python
>>> class Something:
...   def __init__(self):
...     self.a = 33
...
>>> class SomethingForTest(Something):
...   a = 33
...
>>> p = patch('__main__.Something', autospec=SomethingForTest)
>>> mock = p.start()
>>> mock.a
<NonCallableMagicMock name='Something.a' spec='int' id='...'>
```